### PR TITLE
[DOCS] Add common troubleshooting steps to compiling guide

### PR DESCRIPTION
**Pull request by Hundrec**
Due to the recent influx of compiling help issues, I've added a few troubleshooting steps to the Compiling Guide.

This should reduce the amount of work Eric has to do sorting through various compiling help issues.

Any additional troubleshooting steps are appreciated!

### CODE REVIEW

**Review by moxie-coder — APPROVED** [review]

File: docs/COMPILING.md [hunk 1]
```diff
@@ -8,21 +8,23 @@
 1. Run `git clone https://github.com/FunkinCrew/funkin.git` to clone the base repository.
```
Comment (NotHyper-474) [inline docs/COMPILING.md:8]: Very goob, I just think that it should be added that you need to `cd` to the funkin directory before performing step two, or else you get issues like this XD
* #2973

File: docs/COMPILING.md [hunk 2]
```diff
@@ -5,26 +5,28 @@
     - Download Git from [git-scm.com](https://www.git-scm.com)
     - Do NOT download the repository using the Download ZIP button on GitHub or you may run into errors!
     - Instead, open a command prompt and do the following steps...
-1. Run `git clone https://github.com/FunkinCrew/funkin.git` to clone the base repository.
-2. Run `git submodule update --init --recursive` to download the game's assets.
+1. Run `cd the/directory/you/want/the/source/code/in` to specify which folder the command prompt is working in.
+    - For example, `cd C:\Users\YOURNAME\Documents` would instruct the command prompt to perform the next steps in your Documents folder.
```
Comment (NotHyper-474) [inline docs/COMPILING.md]: More specifically, running `git clone...` will create a directory for the repository. I was trying to say that you should put steps that after cloning the repo you should do `cd <repo-folder-name>`, in this case being "funkin" before running `git submodule update...`. You can keep that part, but you should also do something like this:
```diff
- 3. Run `git submodule update --init --recursive` to download the game's assets.
+ 3. Run `cd funkin && git submodule update --init --recursive` to download the game's assets.
```

You can separate the part of doing `cd funkin` in another step if you'd like

Comment (Hundrec) [reply docs/COMPILING.md]: Ah, got it!
I'll make `cd funkin` the new third step and adjust the troubleshooting step accordingly.

**Review by NotHyper-474 — APPROVED** [review]

**Comment by EliteMasterEric** [comment]
Looks good to me!

Since this is a documentation change I'll merge directly into `main`

**Pull request merged**